app/src/main/java/org/wikipedia/analytics/ABTestFunnel.kt

```diff
@@ -0,0 +1,41 @@
+package org.wikipedia.analytics
+
+import org.json.JSONObject
+import org.wikipedia.WikipediaApp
+import org.wikipedia.settings.PrefsIoUtil
+import java.util.*
+
+class ABTestFunnel internal constructor(private val abTestName: String, private val abTestGroupCount: Int) :
+        Funnel(WikipediaApp.getInstance(), SCHEMA_NAME, REV_ID, SAMPLE_LOG_ALL) {
+    // initialize the group if it hasn't been yet.
+    val aBTestGroup: Int
+        get() {
+            var group = PrefsIoUtil.getInt(AB_TEST_KEY_PREFIX + abTestName, -1)
+            if (group == -1) {
+                // initialize the group if it hasn't been yet.
+                group = Random().nextInt(Int.MAX_VALUE)
+                PrefsIoUtil.setInt(AB_TEST_KEY_PREFIX + abTestName, group)
+            }
+            return group % abTestGroupCount
+        }
+    private val isEnrolled: Boolean
+        get() = PrefsIoUtil.contains(AB_TEST_KEY_PREFIX + abTestName)
```

Comment:
Please leave an empty line between variables and it can be simplified as:

`private val isEnrolled = ...`